tests/compat.sh

```diff
@@ -939,13 +939,7 @@ run_client() {
             ;;
 
         [Gg]nu*)
-            # need to force IPv4 with UDP, but keep localhost for auth
```

Comment:
I don't understand why this workaround was needed and I don't understand why it isn't needed anymore. Was it only needed with very old versions of GnuTLS that we no longer test against? Or was it only needed for DTLS 1.0 but not for DTLS 1.2?

Reply:
I _think_ it has to do with GnuTLS version, but it could also be libc version or even local configuration. I really really don't think it has to do with 1.0 vs 1.2.

The problem was that when asked to connect to localhost, at the time GnuTLS would send a datagram to ::1 - but our server was listening on 127.0.0.1 so the two could not communicate. Passing 127.0.0.1 as the host name resolved this but at the cost of breaking cert-based authentication: GnuTLS would then expect "127.0.0.1" in the cert's subject (alt) name while our test certs have "localhost". (The problem was only present for UDP as I assume for TCP GnuTLS would try all the addresses for the hostname until it found one where a server was listening.)